run.py

```diff
@@ -90,8 +79,7 @@
             if points is not None:
                 player.update_score(points)
             if len(targetSprites.sprites()) == 0:
-                shootingTarget = Target()
-                targetSprites.add(shootingTarget)
+                shootingTarget = shootingTarget.spawn_new_target(targetSprites, player)
```

Comment:
This statement made more sense only after I read the function implementation in the target class.  I don't like how the target object requires a group to complete its spawning process.  It's also less self-documenting than how you had the code previously.  Before, it was easy to read that if the group length is 0, then create a new target, and add to the group.  This change makes it harder to read from just looking at the code in this file individually.